pkg/apis/core/types_shoot.go

```diff
@@ -176,6 +176,8 @@ type DNS struct {
 type DNSProvider struct {
 	// Domains contains information about which domains shall be included/excluded for this provider.
 	Domains *DNSIncludeExclude
+	// Primary indicates that this DNSProvider is used for shoot related domains.
+	Primary *bool
```

Comment:
Would it make more sense to have this a not a boolean, but rather something like `Class *DNSProviderClass` with `type DNSProviderClass string`, so later on, we can have multiple options, such as `Primary`, `FailOver` etc...

Reply:
Can make sense, yes.
Although I would rather go with something like `Purpose`. Any thoughts @mvladev @rfranzke?

Reply:
`Purpose` is fine as well.

Reply:
@mvladev I guess what you are hinting could be added as a policy field in the future? Something link: 

```go
// DNS holds information about the provider, the hosted zone id and the domain.
type DNS struct {
	// Domain is the external available domain of the Shoot cluster. This domain will be written into the
	// kubeconfig that is handed out to end-users.
	Domain *string
	// Providers is a list of DNS providers that shall be enabled for this shoot cluster. Only relevant if
	// not a default domain is used.
	Providers []DNSProvider
        // Policy poilicy type e.g., active-passive, fail-over, ... 
        Policy PolicyType 
}
```

To me primary here looks okay.

Reply:
@mvladev I guess what you are hinting could be added as a policy field in the future? Something link: 

```go
// DNS holds information about the provider, the hosted zone id and the domain.
type DNS struct {
	// Domain is the external available domain of the Shoot cluster. This domain will be written into the
	// kubeconfig that is handed out to end-users.
	Domain *string
	// Providers is a list of DNS providers that shall be enabled for this shoot cluster. Only relevant if
	// not a default domain is used.
	Providers []DNSProvider
        // Policy poilicy type e.g., active-passive, fail-over, ... 
        Policy PolicyType 
}
```

To me primary here looks okay.

Reply:
@mvladev I guess what you are hinting could be added as a policy field in the future for DNS configuration? Something like: 

```go
// DNS holds information about the provider, the hosted zone id and the domain.
type DNS struct {
	// Domain is the external available domain of the Shoot cluster. This domain will be written into the
	// kubeconfig that is handed out to end-users.
	Domain *string
	// Providers is a list of DNS providers that shall be enabled for this shoot cluster. Only relevant if
	// not a default domain is used.
	Providers []DNSProvider
        // Policy poilicy type e.g., active-passive, fail-over, ... 
        Policy PolicyType 
}
```

To me primary here looks okay.

Reply:
When I think about that @zanetworker then `primary` totally makes sense, i.e. it will always point out the provider that is currently responsible for the shoot domain.

Reply:
I reiterated over the proposal of using `Purpose *string` instead of `Primary *bool` and think sticking to `Primary` has the following benefits:

1. A shoot will always have only one primary provider, no matter if several other ones are available for fail over, etc.. We can introduce another field like `Policy` (as @zanetworker suggested) if we need to, but the `Primary` purpose will still be valid.
2. `Primary` is easy to use as opposed to an optional string field whose allowed values must be described in a documentation first.